http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/ClusterHelper.scala

```diff
@@ -2,130 +2,222 @@ package org.broadinstitute.dsde.workbench.leonardo
     extends LazyLogging
     with Retry {
 
   def createClusterIamRoles(googleProject: GoogleProject, serviceAccountInfo: ServiceAccountInfo): IO[Unit] =
-    updateClusterIamRoles(googleProject, serviceAccountInfo, true)
+    updateClusterIamRoles(googleProject, serviceAccountInfo, createCluster = true)
 
   def removeClusterIamRoles(googleProject: GoogleProject, serviceAccountInfo: ServiceAccountInfo): IO[Unit] =
-    updateClusterIamRoles(googleProject, serviceAccountInfo, false)
+    updateClusterIamRoles(googleProject, serviceAccountInfo, createCluster = false)
 
-  private def updateClusterIamRoles(googleProject: GoogleProject,
-                                    serviceAccountInfo: ServiceAccountInfo,
-                                    create: Boolean): IO[Unit] = {
-    val retryIam: (GoogleProject, WorkbenchEmail, Set[String]) => IO[Boolean] = (project, email, roles) =>
-      IO.fromFuture[Boolean](IO(retryExponentially(when409, s"IAM policy change failed for Google project '$project'") {
-        () =>
-          if (create) {
-            googleIamDAO.addIamRolesForUser(project, email, roles)
-          } else {
-            googleIamDAO.removeIamRolesForUser(project, email, roles)
+  def generateServiceAccountKey(googleProject: GoogleProject,
+                                serviceAccountEmailOpt: Option[WorkbenchEmail]): IO[Option[ServiceAccountKey]] =
+    // TODO: implement google2 version of GoogleIamDAO
+    serviceAccountEmailOpt.traverse { email =>
+      IO.fromFuture(IO(googleIamDAO.createServiceAccountKey(googleProject, email)))
+    }
+
+  def removeServiceAccountKey(googleProject: GoogleProject,
+                              serviceAccountEmailOpt: Option[WorkbenchEmail],
+                              serviceAccountKeyIdOpt: Option[ServiceAccountKeyId]): IO[Unit] =
+    // TODO: implement google2 version of GoogleIamDAO
+    (serviceAccountEmailOpt, serviceAccountKeyIdOpt).mapN {
+      case (email, keyId) =>
+        IO.fromFuture(IO(googleIamDAO.removeServiceAccountKey(googleProject, email, keyId)))
+    } getOrElse IO.unit
+
+  def setupDataprocImageGoogleGroup(dpImageUserGoogleGroupName: String,
+                                    dpImageUserGoogleGroupEmail: WorkbenchEmail): IO[Boolean] =
+    createDataprocImageUserGoogleGroupIfItDoesntExist(dpImageUserGoogleGroupName, dpImageUserGoogleGroupEmail)
+      .flatMap { _ =>
+        addIamRoleToDataprocImageGroup(dataprocConfig.customDataprocImage, dpImageUserGoogleGroupEmail)
+      }
+
+  /**
+   *  Add the user's service account to the Google group.
+   *  This group has compute.imageUser role on the custom Dataproc image project,
+   *  which allows the user's cluster to pull the image.
+   */
+  def updateDataprocImageGroupMembership(googleProject: GoogleProject, createCluster: Boolean): IO[Unit] = {
+    // TODO Template and get the group and domain values from config
+    val dataprocImageUserGoogleGroupName = "kyuksel-test-dataproc-image-group-11"
```

Comment:
this TODO needs to be addressed before this can be merged

Reply:
yes, there'll be a firecloud-develop PR for that